x-pack/plugins/security_solution/public/management/pages/policy/view/event_filters/hooks.ts

```diff
@@ -35,46 +32,35 @@ export function useGetAllAssignedEventFilters(
 }
 
 export function useSearchAssignedEventFilters(
-  policyId?: string,
-  filter?: string
+  eventFiltersService: EventFiltersHttpService,
+  policyId: string,
+  options: { filter?: string; page?: number; perPage?: number }
 ): QueryObserverResult<FoundExceptionListItemSchema, ServerApiError> {
-  const http = useHttp();
-  const eventFiltersService = new EventFiltersHttpService(http);
+  const { filter, page, perPage } = options;
 
   return useQuery<FoundExceptionListItemSchema, ServerApiError>(
-    ['eventFilters', 'assigned', policyId],
+    ['eventFilters', 'assigned', 'search', policyId, options],
     () => {
-      const kuery = [
-        `((exception-list-agnostic.attributes.tags:"policy:${policyId}") OR (exception-list-agnostic.attributes.tags:"policy:all"))`,
-      ];
-
-      if (filter) {
-        const filterKuery = parseQueryFilterToKQL(filter, SEARCHABLE_FIELDS) || undefined;
-        if (filterKuery) {
-          kuery.push(filterKuery);
-        }
-      }
-
       return eventFiltersService.getList({
-        filter: kuery.join(' AND '),
+        filter: parsePoliciesAndFilterToKql({
+          policies: [policyId, 'all'],
+          kuery: parseQueryFilterToKQL(filter || '', SEARCHABLE_FIELDS),
+        }),
+        perPage,
+        page: (page ?? 0) + 1,
       });
     },
     {
       refetchIntervalInBackground: false,
       refetchOnWindowFocus: false,
-      enabled: !!policyId,
-      refetchOnMount: true,
+      cacheTime: 0, // No cache this query as the search results need to be fresh
```

Comment:
check that this is working as you intended and not creating a new request every time the component renders. you do want some kind of caching for this query and remember you are passing the search and options as the cache key, there's no need to manually handle cache on react-query if you create the correct cache key.

Reply:
Yeah, I know for new cache keys the query will be dispatched. But as this list can be manipulated from other sides of the app (for example from the event filters list) I think is better to dispatch always the query, even if it was cached before. This is to invalidate the query every time we search.

Reply:
in those cases: you should invalidate the cache key in the other side rather than manipulating the query cache directly. Also if the user change tabs, when they focus back on this tab, the query will automatically re-fetch.

In general is better to leave the cache handling to react-query than trying to do it manually. Caching is one of the things react-query does the best.

Reply:
There is no `other side`, basically I don't want to cache this specific query, setting the cache time to 0 is the way to do that, just a simple parameter.

Reply:
Spoke with Esteban offline and this can cause unwanted query fetch when component renders for some other reason.
I'm going to remove this parameter and let react-query play with the cache. Thanks Esteban for the feedback you provided!